hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CellBlocksOffHeap.java

```diff
@@ -0,0 +1,75 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Cellersion 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY CellIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hbase.regionserver;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.KeyValue;
+import org.apache.hadoop.hbase.util.Bytes;
+
+import java.util.Comparator;
+import junit.framework.Assert;
+
+/**
+ * CellBlocksOffHeap is a byte array holding all that is needed to access a Cell, which
+ * is actually saved on another deeper byte array
+ */
+public class CellBlocksOffHeap extends CellBlocks {
+
+  private byte block[];
+  private final HeapMemStoreLAB memStoreLAB;
+
+  public CellBlocksOffHeap(Comparator<? super Cell> comparator, HeapMemStoreLAB memStoreLAB,
+      byte b[], int min, int max, boolean d) {
+    super(comparator,min,max,d);
+    this.block = b;
+    this.memStoreLAB = memStoreLAB;
+  }
+
+  @Override
+  protected CellBlocks createCellBlocks(Comparator<? super Cell> comparator, int min, int max,
+      boolean d) {
+    return new CellBlocksOffHeap(comparator, this.memStoreLAB, this.block, min, max, d);
+  }
```

Comment:
so I see you are not copying the actual array, in this case I must say I am very confused about the semantics of is descending and whether or not methods like tailMap or headMap return the correct answer.
We need a board to see how this works